### PR TITLE
fix: Aggregate viz all time

### LINKED ISSUE #6799: [insights] Pie chart with last24hrs applied shows wrong persons when clicked
## Bug description

*Please describe.*  
- person numbers don't match when you click on a pie chart with last 24 hours applied
- ***They do work when you have the interval set to hourly before you switch to the pie chart

## Expected behavior
- numbers should match

## How to reproduce

1. https://app.posthog.com/insights?insight=TRENDS&actions=%5B%5D&events=%5B%7B%22id%22%3A%22cohort%20created%22%2C%22name%22%3A%22cohort%20created%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%2C%22math%22%3A%22dau%22%7D%5D&properties=%5B%5D&filter_test_accounts=true&exclusions=%5B%5D&funnel_from_step=0&funnel_to_step=1&compare=false&new_entity=%5B%5D&interval=hour&date_from=-24h&display=ActionsLineGraphCumulative

## Environment

- [x] PostHog Cloud
- [x] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- address bug with actions pie #6804 
- should resolve #6799 too
*If this affects the frontend, include screenshots.*  

## How did you test this code?

*Please describe.*
- manual qa

### CODE REVIEW

File: frontend/src/scenes/trends/viz/ActionsPie.tsx [hunk 1]
```diff
@@ -74,8 +74,8 @@ export function ActionsPie({
                             const { dataset } = point
                             const action = dataset.actions[point.index]
                             const label = dataset.labels[point.index]
-                            const date_from = dataset.days[0]
-                            const date_to = dataset.days[dataset.days.length - 1]
+                            const date_from = filtersParam.date_from || ''
```
Comment (hazzadous) [inline frontend/src/scenes/trends/viz/ActionsPie.tsx:77]: non-blocking

What are the cases where `loadPeople` would use the explicitly provided `date_from` and `date_to` over using whatever `filterParams` already provides?

How might I know that `''` is going to do what's expected?

**Review by hazzadous — APPROVED** [review]
Looks plausible although I'm not sure of the details of `loadPeople`, it seems quite complicated. Would you mind walking me through what is going on there?

File: frontend/src/scenes/trends/viz/ActionsPie.tsx (hunk 1 above)
Comment (EDsCODE) [reply frontend/src/scenes/trends/viz/ActionsPie.tsx:77]: loadPeople differs between 2 modes. One a line graph when you're selecting a specific date, you need to specify the exact timeframe among a date range. On a total chart (like pie chart or bar chart by value), you should just provide the filters default and not set it to anything specifically.

> How might I know that '' is going to do what's expected?

This is where the typing you're working on could help. It should make explicit what should always be passed/not passed and what defaults are allowed + what the defaults do. The empty string defers the date control to whatever default we have set up within the date parsing functions when query building

**Comment by EDsCODE** [comment]
> Looks plausible although I'm not sure of the details of `loadPeople`, it seems quite complicated. Would you mind walking me through what is going on there?

loadPeople takes the entire `filters` object along with several extra bits of identifying information that might be necessary such as modified date_from/date_to if you're selecting a specific point, a specific event/action that you're selecting persons on, etc (this structure could be clearer). The listener for that function in the logic is a huge if statement with several cases that chooses the right endpoint based on which insight it is.

**Pull request merged**